src/main/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorImpl.java

```diff
@@ -0,0 +1,189 @@
+package uk.gov.hmcts.darts.dailylist.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.cases.repository.CaseRepository;
+import uk.gov.hmcts.darts.common.entity.CourtCaseEntity;
+import uk.gov.hmcts.darts.common.entity.CourthouseEntity;
+import uk.gov.hmcts.darts.common.entity.DailyListEntity;
+import uk.gov.hmcts.darts.common.entity.HearingEntity;
+import uk.gov.hmcts.darts.common.entity.JudgeEntity;
+import uk.gov.hmcts.darts.common.exception.DartsApiException;
+import uk.gov.hmcts.darts.common.repository.DefenceRepository;
+import uk.gov.hmcts.darts.common.repository.DefendantRepository;
+import uk.gov.hmcts.darts.common.repository.HearingRepository;
+import uk.gov.hmcts.darts.common.repository.ProsecutorRepository;
+import uk.gov.hmcts.darts.common.service.RetrieveCoreObjectService;
+import uk.gov.hmcts.darts.courthouse.CourthouseRepository;
+import uk.gov.hmcts.darts.dailylist.enums.JobStatusType;
+import uk.gov.hmcts.darts.dailylist.enums.SourceType;
+import uk.gov.hmcts.darts.dailylist.exception.DailyListError;
+import uk.gov.hmcts.darts.dailylist.model.CitizenName;
+import uk.gov.hmcts.darts.dailylist.model.CourtList;
+import uk.gov.hmcts.darts.dailylist.model.DailyList;
+import uk.gov.hmcts.darts.dailylist.model.Defendant;
+import uk.gov.hmcts.darts.dailylist.model.Hearing;
+import uk.gov.hmcts.darts.dailylist.model.Sitting;
+import uk.gov.hmcts.darts.dailylist.repository.DailyListRepository;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class DailyListProcessorImpl implements DailyListProcessor {
+    private final DailyListRepository dailyListRepository;
+    private final RetrieveCoreObjectService retrieveCoreObjectService;
+    private final CourthouseRepository courthouseRepository;
+    private final CaseRepository caseRepository;
+    private final HearingRepository hearingRepository;
+    private final DefendantRepository defendantRepository;
+    private final DefenceRepository defenceRepository;
+    private final ProsecutorRepository prosecutorRepository;
+
+    private final ObjectMapper objectMapper;
+
+    @Override
+    public void processAllDailyLists(LocalDate date) {
+        List<CourthouseEntity> allCourthouses = courthouseRepository.findAll();
+        for (CourthouseEntity allCourthouse : allCourthouses) {
+            List<DailyListEntity> dailyLists = getDailyListForCppAndXhb(date, allCourthouse);
```

Comment:
i dont think this code change has had the desired effect. it still looks like it gets all the dailylists for a courthouse, then only processes 1, whereas we would want to process the latest one for each sourceType.

Reply:
ACCEPT.